discrimination-ieee754/test/test.hs

```diff
@@ -0,0 +1,23 @@
+module Main where
+
+import Data.Discrimination (Sort, runSort)
+import Data.Discrimination.IEEE754 (sortingFloat, sortingDouble)
+
+import Test.Framework (defaultMain)
+import Test.Framework.Providers.API (Test)
+import Test.Framework.Providers.QuickCheck2 (testProperty)
+
+sortCompare :: Sort a -> a -> a -> Ordering
+sortCompare s l r = case runSort s [(l, LT), (r, GT)] of
+    [x]:_ -> x
+    _     -> EQ
+
+propAgreesWithOrd :: Ord a => Sort a -> (a, a) -> Bool
+propAgreesWithOrd s (l, r) = compare l r == sortCompare s l r
+
+main = defaultMain
```

Comment:
It turns out Ord is somewhat nonsensical for Float and Double because the boolean comparisons obey the requirement that NaN is incomparable to anything, and 'compare' is implemented in terms of them.  So `compare NaN x` is `GT`, but `compare x NaN` is also `GT`.  So I short-circuited out the "agree with Ord" tests for NaN arguments, added a bunch of special cases, and added tests that this matches the IEEE754 total ordering to cover the NaN cases.